cumulus/client/consensus/aura/src/collators/basic.rs

```diff
@@ -246,11 +246,7 @@ where
 						None,
 						(parachain_inherent_data, other_inherent_data),
 						params.authoring_duration,
-						// Set the block limit to 50% of the maximum PoV size.
-						//
-						// TODO: If we got benchmarking that includes the proof size,
-						// we should be able to use the maximum pov size.
-						(validation_data.max_pov_size / 2) as usize,
+						validation_data.max_pov_size,
```

Comment:
AFAIR this is safe to do now with `on_idle` fix.

Reply:
We can't do that in this PR this easily. This is a client-side change and the fix you did with `CheckWeight` will be a runtime change. So if someone uses this collator with for example kusama before the runtime there is upgraded to use the new `CheckWeight`, this bug can happen.

Reply:
Yes, this will have to wait ...